### PR TITLE
Canonicalize metrics docs

### PR DESCRIPTION
**Pull request by WindQAQ**
# Description

Canonicalize metrics docs.

For testable docstring:

- Naming convention: `y_true`, `y_pred`, `metric`, `result`.
- Use numpy array, which is the same with core TF.

For doc rendering:

- Put `Args` section between `class` and `__init__`, which is the same with core TF.
- Add `name` and `dtype` doc if applicable.

## Type of change

- [ ] Bug fix
- [ ] New Tutorial
- [x] Updated or additional documentation
- [ ] Additional Testing
- [ ] New Activation and the changes conform to the [activation contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/activations/README.md#contribution-guidelines)
- [ ] New Callback and the changes conform to the [callback contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/callbacks/README.md#contribution-guidelines)
- [ ] New Image addition and the changes conform to the [image op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/image/README.md#contribution-guidelines)
- [ ] New Layer and the changes conform to the [layer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/layers/README.md#contribution-guidelines)
- [ ] New Loss and the changes conform to the [loss contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/README.md#contribution-guidelines)
- [ ] New Metric and the changes conform to the [metric contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/README.md#contribution-guidelines)
- [ ] New Optimizer and the changes conform to the [optimizer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/README.md#contribution-guidelines)
- [ ] New RNN Cell and the changes conform to the [rnn contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/rnn/README.md#contribution-guidelines)
- [ ] New Seq2seq addition and the changes conform to the [seq2seq contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/seq2seq/README.md#contribution-guidelines)
- [ ] New Text addition and the changes conform to the [text op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/text/README.md#contribution-guidelines)

# Checklist:

- [x] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    - [ ] By running Black + Flake8
    - [ ] By running pre-commit hooks
- [ ] This PR addresses an already submitted issue for TensorFlow Addons
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] This PR contains modifications to C++ custom-ops

# How Has This Been Tested?

If you're adding a bugfix or new feature please describe the tests that you ran to verify your changes:
*

### CODE REVIEW

File: tensorflow_addons/metrics/hamming.py [hunk 1]
```diff
@@ -30,20 +30,19 @@ def hamming_distance(actuals: TensorLike, predictions: TensorLike) -> tf.Tensor:
...
+        actuals: actual target value.
+        predictions: predicted value.
 
     Returns:
-        hamming distance: float
+        hamming distance: float.
 
     Usage:
 
     >>> actuals = tf.constant([1, 1, 0, 0, 1, 0, 1, 0, 0, 1],
```
Comment (AakashKumarNain) [inline tensorflow_addons/metrics/hamming.py:41]: This should also be changed to `y_true` and `y_pred`

File: tensorflow_addons/metrics/hamming.py [hunk 2]
```diff
@@ -30,20 +30,19 @@ def hamming_distance(actuals: TensorLike, predictions: TensorLike) -> tf.Tensor:
...
     ... dtype=tf.int32)
     >>> predictions = tf.constant([1, 0, 0, 0, 1, 0, 0, 1, 0, 1],
     ... dtype=tf.int32)
-    >>> metric = hamming_distance(actuals, predictions)
-    >>> metric.numpy()
+    >>> hamming_distance(actuals, predictions).numpy()
```
Comment (AakashKumarNain) [inline tensorflow_addons/metrics/hamming.py:45]: Same

**Comment by AakashKumarNain** [comment]
Thanks @WindQAQ Suggested one more change. Apart from that LGTM!

File: tensorflow_addons/metrics/hamming.py (hunk 2 above)
Comment (WindQAQ) [reply tensorflow_addons/metrics/hamming.py]: Done :-)

Comment (AakashKumarNain) [reply tensorflow_addons/metrics/hamming.py]: Thanks

**Review by AakashKumarNain — APPROVED** [review]

**Pull request merged**